### PR TITLE
Modified diags.py,adding more read observatio items, the data categor…

### PR DESCRIPTION
**Pull request by XiujuanSu-NOAA**
See the reference  to issue  #59  and  #81460

### CODE REVIEW

File: pyGSI/diags.py [hunk 1]
```diff
@@ -234,28 +246,37 @@ def get_data(self, diag_type, obsid=None, subtype=None, station_id=None, analysi
...
+                           rejected_idx[0], hght_idx[0])
                        valid_monitored_idx = np.isin(
                            monitored_idx[0], hght_idx[0])
                     else:
                        pres_idx = np.where((self.press > level_list[i]) & (
                            self.press <= level_list[i+1]))
                        valid_assimilated_idx = np.isin(
                            assimilated_idx[0], pres_idx[0])
+                       valid_rejected_idx = np.isin(
+                        rejected_idx[0], pres_idx[0])
```
Comment (CoryMartin-NOAA) [inline pyGSI/diags.py:259]: Can you fix the indentation of this line so that it matches the others?

**Review by CoryMartin-NOAA — CHANGES_REQUESTED** [review]
One very minor change, otherwise I think it looks good

**Review by EdwardSafford-NOAA — DISMISSED** [review]
Looks good.

**Comment by XiujuanSu-NOAA** [comment]
Sure Thanks

Su

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**